src/bindings/jna/CMakeLists.txt

```diff
@@ -23,136 +30,166 @@ if (Java_JAVAC_EXECUTABLE)
-				${JNA_BINDING_SOURCE_DIRECTORY_TEST}/KeyTest.java)
-
-			# Compile the source files and build the jar. We copy the source files into the build folder as that is the easiest
-			# way to handle the integration between CMake and Maven.
-			add_custom_command (
-				OUTPUT "${CMAKE_CURRENT_BINARY_DIR}/libelektra5j/target/libelektra5j-${KDB_VERSION}.jar"
-				COMMAND ${CMAKE_COMMAND} ARGS -E copy_directory "${CMAKE_CURRENT_SOURCE_DIR}/libelektra5j"
-					"${CMAKE_CURRENT_BINARY_DIR}/libelektra5j"
-				COMMAND ${MAVEN_EXECUTABLE} -q -B -DskipTests=true package
-				WORKING_DIRECTORY libelektra5j
-				DEPENDS "${CMAKE_CURRENT_BINARY_DIR}/libelektra5j/pom.xml" ${JNA_BINDING_SOURCE_FILES})
-
-			create_lib_symlink (libelektra5j-${KDB_VERSION}.jar libelektra5j.jar java-elektra JAVA)
-			create_lib_symlink (libelektra5j-${KDB_VERSION}.pom.xml libelektra5j.pom.xml java-elektra JAVA)
-
-			add_custom_target (jna ALL
-					   DEPENDS "${CMAKE_CURRENT_BINARY_DIR}/libelektra5j/target/libelektra5j-${KDB_VERSION}.jar")
-
-			install (
-				FILES "${CMAKE_CURRENT_BINARY_DIR}/libelektra5j/target/libelektra5j-${KDB_VERSION}.jar"
-				      "${CMAKE_CURRENT_BINARY_DIR}/libelektra5j/libelektra5j-${KDB_VERSION}.pom.xml"
-				DESTINATION "share/java"
-				COMPONENT java-elektra)
-
-			# and add maven test to execute with ctest in the testing phase
-			if ((CMAKE_SYSTEM_NAME STREQUAL "Darwin") AND NOT (CMAKE_SYSTEM_VERSION VERSION_LESS 15))
-
-				# we cannot set DYLD_LIBRARY_PATH on new macOS versions, making the kdb tests fail if its not installed in
-				# the system
-				set (MAVEN_TEST_ENV "env-test-without-kdb")
+		endif ()
+
+		if (GRADLE_EXECUTABLE) # set by find_program
+			# check Gradle version
+			execute_process (
+				COMMAND bash "-c" "${GRADLE_EXECUTABLE} -v | grep 'Gradle *' | egrep -o '[0-9]+\.[0-9]+(\.[0-9]+)?'"
+				OUTPUT_VARIABLE GRADLE_VERSION
+				OUTPUT_STRIP_TRAILING_WHITESPACE)
+
+			if ((${GRADLE_VERSION} VERSION_GREATER ${MIN_GRADLE_VERSION}) OR (${GRADLE_VERSION} VERSION_EQUAL
+											  ${MIN_GRADLE_VERSION}))
+				add_binding (jna)
+
+				file (MAKE_DIRECTORY libelektra)
+
+				# generate test exclusions based on specific plugins and environment
+				set (GRADLE_TEST_EXCLUDES "")
+				java_test_needs_plugin (GOptsTest gopts)
+
+				if ((CMAKE_SYSTEM_NAME STREQUAL "Darwin") AND NOT (CMAKE_SYSTEM_VERSION VERSION_LESS 15))
+					# we cannot set DYLD_LIBRARY_PATH on new macOS versions, making the kdb tests fail if its not
+					# installed in the system
+					java_exclude_test (ExceptionMapperIT)
+					java_exclude_test (ExceptionMapperTest)
+					java_exclude_test (GOptsTest)
+					java_exclude_test (KDBExceptionTest)
+					java_exclude_test (KDBTest)
+					java_exclude_test (KeyNameIteratorTest)
+					java_exclude_test (KeySetTest)
+					java_exclude_test (KeyTest)
+					java_exclude_test (PluginLoaderIT)
+					java_exclude_test (ReturnTest)
+					java_exclude_test (WarningEntryTest)
+				endif ()
```

Comment:
Test exclusion did not work at all until now, so I suspect this code had no effect. If so, I'd like to get rid of that after the release.

Reply:
Possibly, this is no longer required with the new setup. But with the old Maven setup, tests that need the KDB definitely did not work under macOS.

Reply:
What exactly did not work with the test exclusions?

Reply:
It is fixed now so no need to do anything, but the problem was this line:
https://github.com/tucek/libelektra/blob/dc6914eca9ed1a9929f93ef7d650288a58622eb2/src/bindings/jna/CMakeLists.txt#L14

Basically the static string `"test"` is passed instead of the variable `${test}`. I took a look at the generated `test-exclusion.gradle` and confirmed that.